legion/engine/llri-vk/utils.cpp

```diff
@@ -0,0 +1,119 @@
+#include <llri/llri.hpp>
+#include <vulkan/vulkan.hpp>
+#include <map>
+#include <string>
+#include <vector>
+
+namespace legion::graphics::llri
+{
+    namespace internal
+    {
+        /**
+         * @brief Helper function that maps layers to their names
+        */
+        std::map<std::string, vk::LayerProperties> preprocessLayerProperties()
+        {
+            std::map<std::string, vk::LayerProperties> result;
```

Comment:
any particular reason for these layers to be alphabetically sorted? maybe you wanted to use an `std::unordered_map`?
also are you sure you don't want to use prehashed or compiletime hashed strings instead of actual strings? strings in maps need to be rehashed on every operation.

Reply:
No, I thought ordered maps were more performant. I'm not sure if these can be compile time hashed because the names are queried in runtime

Reply:
ordered maps are actually less performant because they need to sort.

Reply:
@LeonBrands